packages/side-nav/src/SideNavGroup.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { ulStyleOverrides, LEFT_RIGHT_OFFSET } from './styles';
+
+const sideNavLabelStyle = css`
+  color: ${uiColors.green.base};
+  text-transform: uppercase;
+  font-size: 12px;
+  margin-top: 0;
+  margin-bottom: 0;
+  padding: 5px ${LEFT_RIGHT_OFFSET}px 5px ${LEFT_RIGHT_OFFSET}px;
+`;
+
+interface SideNavGroupProps {
+  /**
+   * Class name that will be applied to the root-level element.
+   */
+  className?: string;
+  /**
+   * Text that will be rendered as the component's header.
+   */
+  headerText?: string;
+  /**
+   * Class name that will be applied to the component's header.
+   */
+  headerClassName?: string;
+  /**
+   * Content that will be rendered inside the root-level element.
+   */
+  children?: React.ReactNode;
+}
+
+/**
+ * # SideNavGroup
+ *
+ * ```
+<SideNavGroup headerText="Section Header">
+  <SideNavItem href="/">
+    Back to Home
+  </SideNavItem>
+</SideNavGroup>
+ * ```
+ * @param props.className Class name that will be applied to the root-level element.
+ * @param props.headerText Text that will be rendered as the component's header.
+ * @param props.headerClassName Class name that will be applied to the component's header.
+ * @param props.children Class name that will be applied to the component's header.
+ *
+ */
+function SideNavGroup({
+  headerText,
+  headerClassName,
+  children,
+  ...rest
+}: SideNavGroupProps) {
+  return (
+    <li {...rest}>
+      {headerText && (
```

Comment:
In an effort to mildly premature optimize.....I wonder if we should rename this prop to `header` and allow it to to be any valid react node type. If it's a string it does what's here, but if it's not then it just renders itself.

Thinking is instead of us someday adding a `tooltipNode` prop in the future we just expose that additional control. 

Happy to be pushed back on this, @DesignerDave usually does :).

Reply:
I wanted to do this, and opted not to because of premature optimization concerns when I was trying to get this out the door. I agree though and am going to change this. It also simplifies our props, as we don't need to take a custom class anymore when you can just directly pass in the header.

I will have to think a little bit about this and ensuring there is either an `h*` tag or `role="header"` somewhere.